File: Graph Algorithms in Genome Sequencing/Genome seq/Generate All Maximal Non-Branching Paths in a Graph/max_non_branching.cpp
#include <iostream>
#include <vector>
#include <string>

using std::cout;
using std::cin;
using std::endl;
using std::string;
using std::vector;
using std::getline;


/*
 * Finds all maximum non-branching paths in the given graph
 *
 * \param graph     a vector of strings containing edges in the graph;
 *                  one string may correspond to multiple edges if it
 *                  is of the form "X -> Y,Z"
 *
 * \return a string containing all maximum non-branching paths in the
 * format specified by the problem.
 */
string find_paths(const vector<string>& graph) {
    // TODO: your code here
    return "";
}


int main(int argc, char* argv[]) {
    vector<string> graph;

    string line;
    while (getline(cin, line, '\n')) {
        graph.push_back(line);
    }

    cout << find_paths(graph) << endl;
}

File: Graph Algorithms in Genome Sequencing/Genome seq/Generate All Maximal Non-Branching Paths in a Graph/MaxNonBranching.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.List;

public class MaxNonBranching {
    /*
     * Finds all maximum non-branching paths in the given graph
     *
     * @param graph     a List of Strings containing edges in the graph;
     *                  one String may correspond to multiple edges if it
     *                  is of the form "X -> Y,Z"
     *
     * @return a String containing all maximum non-branching paths in the
     * format specified by the problem
     */
    public static String findPaths(List<String> graph) {
        // TODO: your code here
        return "";
    }

    public static void main(String[] args) {
        Scanner stdin = new Scanner(System.in);
        List<String> graph = new ArrayList<String>();

        while (stdin.hasNextLine()) graph.add(stdin.nextLine());

        System.out.println(findPaths(graph));
    }
}
